File: nodejs/src/factory/products/ItalianCoffee.ts
class ItalianCoffee {
    prepare() {
        return "Pha Espresso kiểu Ý";
    }
    getPrice() {
        return 55000;
    }
    getDescription() {
        return "Espresso";
    }
    getCost() {
        return this.getPrice();
    }
}

export default ItalianCoffee;

File: spring/demo/src/main/java/com/example/demo/factory/products/ItalianCoffee.java
package com.example.demo.factory.products;

public class ItalianCoffee implements Coffee {
    @Override public String prepare() { return "Pha Espresso kiểu Ý"; }
    @Override public int getPrice() { return 55000; }
    @Override public String getDescription() { return "Espresso"; }
    @Override public int getCost() { return getPrice(); }
}
